cf/field.py

```diff
@@ -1707,27 +1730,33 @@ def _equivalent_construct_data(
         item1 = field1.constructs[key1]
 
         if item0.has_data() != item1.has_data():
-            logger.info(
-                f"{self.__class__.__name__}: Only one item has data"
-            )  # pragma: no cover
+            if is_log_level_info(logger):
```

Comment:
Hi Sadie, I think that it's fastest left as it is, because each one comes before a `return False`, so at most only one is ever executed, and then only if the equivalence is False. If we put in a call at the start, then it would get run when the equivalence is True also.

Reply:
Fair enough, that makes sense now you reason it through. Thanks, marking as resolved.